### PR TITLE
refactor(tx): tx management consolidation

### PR DESCRIPTION
**Pull request by noandrea**
Issues #238 #237 #239

### CODE REVIEW

File: aeternity/transactions.py [hunk 1]
```diff
@@ -1,40 +1,432 @@
-from aeternity.hashing import _int, _int_decode, _binary, _binary_decode, _id, _id_decode, encode, decode, encode_rlp, decode_rlp, hash_encode
+from aeternity.hashing import _int, _int_decode, _binary, _binary_decode, _id, _id_decode, encode, decode, hash_encode
 from aeternity.openapi import OpenAPICli
 from aeternity import identifiers as idf
 from aeternity import defaults
-from aeternity.exceptions import UnsupportedTransactionType, TransactionFeeTooLow
+# from aeternity.exceptions import UnsupportedTransactionType, TransactionFeeTooLow
```
Comment (shekhar-shubhendu) [inline aeternity/transactions.py]: unused imports can be removed

File: aeternity/transactions.py [hunk 2]
```diff
@@ -1,40 +1,432 @@
...
+
+    def get_signatures(self):
+        """
+        retrieves the list of signatures for a signed transaction, otherwhise returns a empty list
+        """
+        if self.data.tag == idf.OBJECT_TAG_SIGNED_TRANSACTION:
+            return self.data.signatures
+        return []
+
+    def get_signature(self, index):
```
Comment (shekhar-shubhendu) [inline aeternity/transactions.py]: can be merged into a single function with `index` as optional param

File: aeternity/transactions.py [hunk 3]
```diff
@@ -1000,19 +482,226 @@ def compute_tx_hash(encoded_tx: str) -> str:
...
+                # this are name pointers
+                tx_data[label] = [{"key": _binary_decode(p[0], data_type=str), "id": _id_decode(p[1])} for p in raw[fn.index]]
+            elif fn.field_type == _TX:
+                # this can be raw or tx object
+                tx_data[label] = self._rlptx_to_txobject(raw[fn.index])
+        # re-encode the decode object
+        compute_hash = True if tag == idf.OBJECT_TAG_SIGNED_TRANSACTION else False
+        return self._txdata_to_txobject(tx_data, descriptor, compute_hash=compute_hash)
+
+    def _build_tx_object(self, tx_data, metadata={}) -> TxObject:
```
Comment (shekhar-shubhendu) [inline aeternity/transactions.py]: should be `_build_txobject`

Comment (noandrea) [reply aeternity/transactions.py]: fixed

**Comment by shekhar-shubhendu** [comment]
🏆 💯 🌟 🥇

**Pull request merged**